quil-rs/src/parser/common.rs

```diff
@@ -195,6 +195,60 @@ pub(crate) fn parse_permutation<'a>(input: ParserInput<'a>) -> InternalParserRes
     )(input)
 }
 
+/// Parse a collection of [`crate::instruction::PauliWord`]s. These are a special kind of a
+/// [`Token::Identifier`] used in Pauli sum `DEFGATE` specifications where each identifier
+/// is a single string made up of one or more `I`, `X`, `Y`, `Z` characters (e.g. `X`, `YY`),
+/// each representing a Pauli word.
```

Comment:
This is the simplest approach I could come up with. A new `Token` variant for Pauli words won't work, because it would have to be parsed before a `Command`/`Identifier`, but the individual characters overlap with Reserved `Identifier`s (Each of the Pauli words correspond to the standard gate of the same name).